qpm v0.10.0 (new formula)

jrbarron opened the pull request:
qpm is a package manager designed for managing dependencies of Qt/QML (https://www.qt.io) applications.

DomT4 commented on Library/Formula/qpm.rb:
```diff
@@ -0,0 +1,18 @@
+class Qpm < Formula
+  desc "Package manager for Qt applications"
+  homepage "https://www.qpm.io"
+  url "https://github.com/Cutehacks/qpm.git", :tag => "v0.10.0",
```
Would you mind putting the `tag` on its own line? Thanks. Trying to keep URL/tag/revision lines more or less in sync so people can just glance down a formula rather than quite far across.

DomT4 commented on Library/Formula/qpm.rb:
```diff
@@ -0,0 +1,19 @@
+
+  def install
+    ENV["GOPATH"] = buildpath
+    system "go", "install", "qpm.io/qpm"
```
We prefer to vendor submodules in ourselves using `go_resource`s and then use `go build` instead of `go install`.

Seems to be pulling a lot in:

```
Submodule 'src/github.com/golang/protobuf' (https://github.com/golang/protobuf.git) registered for path 'src/github.com/golang/protobuf'
Submodule 'src/github.com/howeyc/gopass' (https://github.com/howeyc/gopass) registered for path 'src/github.com/howeyc/gopass'
Submodule 'src/golang.org/x/crypto' (https://go.googlesource.com/crypto) registered for path 'src/golang.org/x/crypto'
Submodule 'src/golang.org/x/net' (https://go.googlesource.com/net) registered for path 'src/golang.org/x/net'
Submodule 'src/google.golang.org/grpc' (https://github.com/grpc/grpc-go) registered for path 'src/google.golang.org/grpc'
Cloning into 'src/github.com/golang/protobuf'...
```

jrbarron commented:
@mikemcquaid I've updated the formula to install a package now, but it is failing in Travis due to a failing assertion about a file existing. Running `brew test --verbose qpm` works fine locally and strangely enough, when I added a system call to `ls -al` to the test before the assertion, it passed in Travis. Any ideas?

DomT4 commented:
Seems to be a transient issue that doesn't consistently reproduce, even on Travis. Will ignore it for now; can always review as it passes through the CI each time.

Merged in https://github.com/Homebrew/homebrew/commit/c34ee713598563a2d32563f3b90da93de2b13965. Thank you for your contribution to Homebrew @jrbarron; we appreciate it! :smiley_cat:
